lambda.tf

```diff
@@ -1,7 +1,63 @@
-data "aws_lambda_function" "set_prerender_header" {
-  function_name = "crest-prerender-test-SetPrerenderHeader-TtxzCLiZqbo6"
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com", "edgelambda.amazonaws.com",]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name               = "lambda-execution_${var.project}_${var.locale}"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "archive_file" "prerender_headers" {
+  type        = "zip"
+  source_file = "${path.module}/prerender-headers.js"
+  output_path = "${path.module}/prerender-headers.zip"
+}
+
+data "archive_file" "redirect_to_prerender" {
+  type        = "zip"
+  source_file = "${path.module}/redirect.js"
+  output_path = "${path.module}/redirect.zip"
 }
 
-data "aws_lambda_function" "redirect_to_prerender" {
-  function_name = "crest-prerender-test-RedirectToPrerender-FBhJ3KhHMR9K"
+resource "aws_lambda_function" "prerender_headers" {
+  filename      = "${path.module}/prerender-headers.zip"
+  function_name = "set-prerender-headers_${var.project}"
+  role          = aws_iam_role.iam_for_lambda.arn
+
+  source_code_hash = data.archive_file.prerender_headers.output_base64sha256
+
+  runtime = "nodejs14.x"
```

Comment:
Why not use the latest Node LTS?

Reply:
You are correct. Even more so as the 14.x runtime is deprecated as of 2023-11-27.